### PR TITLE
client: refine the log errs in client

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

To solve #2704 

### Check List

Tests

- Unit test
- Integration test

### Release note

- No release note

### CODE REVIEW

**Comment by JmPotato** [comment]
@ZenoTan PTAL

File: client/base_client.go [hunk 1]
```diff
@@ -137,7 +138,7 @@ func (c *baseClient) leaderLoop() {
 		}
 
 		if err := c.updateLeader(); err != nil {
-			log.Error("[pd] failed updateLeader", zap.Error(err))
+			log.Error("[pd] failed updateLeader", zap.Error(errs.ErrGRPCSend.FastGenByArgs()), zap.NamedError("cause", err))
```
Comment (ZenoTan) [inline client/base_client.go:141]: Update leader error may not be grpc send error. What about creating a new error type to solve the function with too many possible error sources?

Comment (JmPotato) [reply client/base_client.go:141]: You're right. Already working on it.

**Comment by ZenoTan** [comment]
LGTM

**Pull request merged**